fair/views.py

```diff
@@ -39,9 +39,10 @@ def login_redirect(request):
 
 
 def index(request, year=None):
-    fair = Fair.objects.filter(current=True).first()
+    fair = Fair.objects.filter(year=year).first()
     if fair is None:
-        fair = get_object_or_404(Fair, year=year)
+        fair = Fair.objects.filter(current=True).first()
+        return redirect('fair:home', fair.year)
 
```

Comment:
This change made it possible for redirecting the user to the current fair if they were entering an other fair year in the url, this would fall them back to the current year. This is possible because of a fair by the user input has already created and can be found in the database.